src/irc_quotegrabs.lua

```diff
@@ -0,0 +1,148 @@
+local sql = require 'lsqlite3'
+
+local wl_ins = nil
+local wl_del = nil
+local wl_sel = nil
+
+local qg_ins = nil
+local qg_sel_nick = nil
+local qg_sel_id = nil
+local qg_sel_rand = nil
+local qg_search = nil
+local qg_del = nil
+
+local quotegrabs = {}
+
+-- table to keep track of each nick's last message
+quotegrabs.last_msgs = {}
+
+quotegrabs.whitelist_status = function (nick)
+    wl_sel:reset()
+    wl_sel:bind_names({ ['nick'] = nick })
+    for c in wl_sel:urows() do
+        return c > 0
+    end
+end
+
+quotegrabs.whitelist_add = function (nick)
+    wl_ins:reset()
+    wl_ins:bind_names({ ['nick'] = nick })
+    local res = wl_ins:step()
+    return (res == 101 and 'Tada!' or db:errmsg())
```

Comment:
as was just done in `irc_factoids`, `101` should be replaced with `sql.DONE`. same elsewhere in this file.